enumerable_methods.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  # my each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = is_a?(Enumerable) && !is_a?(Array) ? to_a : self
+    counter = 0
+    while counter < size
+      yield (arr[counter])
+      counter += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = is_a?(Enumerable) && !is_a?(Array) ? to_a : self
+    counter = 0
+    while counter < size
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  # my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_all?(*arg)
+    return grep(arg.first).length == size unless arg.empty?
+
+    my_each { |result| return false unless yield(result) } if block_given?
+    my_each { |result| return false unless result } unless block_given?
+    true
+  end
+
+  # my_any?
+  def my_any?(*arg)
+    return !grep(arg.first).empty? unless arg.empty?
+
+    my_each { |result| return true if yield(result) } if block_given?
+    my_each { |result| return true if result } unless block_given?
+    false
+  end
+
+  # my_none
+  def my_none?(*arg)
+    return grep(arg.first).empty? unless arg.empty?
+
+    my_each { |result| return false if yield(result) } if block_given?
+    my_each { |result| return false if result } unless block_given?
+    true
+  end
+
+  # my_count
+  def my_count(*arg, &block)
+    return my_select { |result| result == arg.first }.length unless arg.empty?
+    return my_select(&block).length if block_given?
+
+    size
+  end
+
+  # my_map
+  def my_map(par_ = nil)
+    new_arr = []
+    return to_enum unless block_given?
+
+    if par_
+      my_each { |i| new_arr << par_.call(i) }
+    else
+      my_each { |i| new_arr << yield(i) }
+    end
+    new_arr
+  end
+
+  # my_inject
+  def my_inject(*arg)
+    res = nil
```

Comment:
[Please fix] Exactly after defining the my_inject function you need to throw a LocalJumpError if there is no block given or no argument: `raise LocalJumpError if !block_given? && arg.empty?`. Then if you call the my_inject on an array without passing a block or an argument you receive a LocalJumpError exactly like the original `inject` method does.